src/modules/profile/sagas.ts

```diff
@@ -115,30 +115,22 @@ export function createProfileSaga({ peerUrl }: CreateProfileSagaOptions) {
     address: string,
     changes: Partial<Avatar>
   ) {
-    const entity: ProfileEntity = yield call(
-      [entities, entities.getProfileEntity],
-      address
-    )
+    const profile: Profile = yield call([peerApi, 'fetchProfile'], address)
+    const profileWithContentHashes = lambdaProfileToContentProfile(profile)
 
     const newAvatar: Avatar = {
-      ...entity.metadata.avatars[0],
+      ...profileWithContentHashes.avatars[0],
       ...changes,
-      version: entity.metadata.avatars[0].version + 1
+      version: profileWithContentHashes.avatars[0].version + 1
     }
 
     const profileMetadata: Profile = {
-      ...entity.metadata,
```

Comment:
I would still keep the data from the metadata in case they add something here, so we don't break the profile metadata, what do you think?

Reply:
there's no `metadata` in the profile fetched from the lambda, it only has the `timestamp` and `avatars` object. Ref: https://github.com/decentraland/catalyst/blob/main/lambdas/src/apis/profiles/controllers/profiles.ts#L379-L387C2